.azure-pipelines.yml

```diff
@@ -0,0 +1,248 @@
+        sccache --show-stats
+
+    - script: cargo fetch
+      displayName: Fetch cargo dependencies
+    - script: cargo fmt --all -- --check
+      displayName: Check formatting
+      condition: and(eq(variables['rustup_toolchain'], 'stable'), ne( variables['build_type'], 'release' ),
+                     eq(variables['Agent.OS'], 'Darwin'))
+
+    - task: CacheBeta@0
+      inputs:
+        key: |
+          target
+          $(Agent.OS)
+          $(RUSTC_VERSION)
+          $(build_type)
+          $(Build.SourcesDirectory)/Cargo.lock
+        path: $(SCCACHE_DIR)
+      displayName: sccache
+
+    # Build and test all features except for lightbeam
+    - script: cargo build --all --exclude lightbeam --exclude wasmtime-wasi-c
+      displayName: Cargo build debug
+      condition: ne( variables['build_type'], 'release' )
+    - bash: RUST_BACKTRACE=1 cargo test --all --exclude lightbeam --exclude wasmtime-wasi-c
+      displayName: Cargo test debug
+      condition: ne( variables['build_type'], 'release' )
+    - script: cargo build --release --all --exclude lightbeam --exclude wasmtime-wasi-c
+      displayName: Cargo build release
+      condition: eq( variables['build_type'], 'release' )
+    - bash: RUST_BACKTRACE=1 cargo test --release --all --exclude lightbeam --exclude wasmtime-wasi-c
+      displayName: Cargo test release
+      condition: eq( variables['build_type'], 'release' )
+
+    # Build and test lightbeam if we're using the nightly toolchain
+    - script: cargo build --package lightbeam
+      displayName: Cargo build lightbeam debug
+      condition: and(eq(variables['rustup_toolchain'], 'nightly'), ne( variables['build_type'], 'release' ))
+    - bash: RUST_BACKTRACE=1 cargo test --package lightbeam
+      displayName: Cargo test lightbeam debug
+      # Lightbeam tests fail right now, but we don't want to block on that.
+      continueOnError: true
+      condition: and(eq(variables['rustup_toolchain'], 'nightly'), ne( variables['build_type'], 'release' ))
+    - script: cargo build --release --package lightbeam
+      displayName: Cargo build lightbeam release
+      condition: and(eq(variables['rustup_toolchain'], 'nightly'), eq( variables['build_type'], 'release' ))
+    - bash: RUST_BACKTRACE=1 cargo test --release --package lightbeam
+      displayName: Cargo test lightbeam release
+      # Lightbeam tests fail right now, but we don't want to block on that.
+      continueOnError: true
+      condition: and(eq(variables['rustup_toolchain'], 'nightly'), eq( variables['build_type'], 'release' ))
+
+    - script: sccache --show-stats
+
+    # Only run on macOS builders, as those are fastest
+    - bash: |
+        cargo doc
+      displayName: Test docs generation
+      condition: and(eq( variables['Agent.OS'], 'Darwin' ), eq( variables['build_type'], 'release' ))
+
+    - bash: |
+        mkdir $BUILD_BINARIESDIRECTORY/bundle
+        cp target/release/{wasmtime,wasm2obj}.exe $BUILD_BINARIESDIRECTORY/bundle
```

Comment:
This may be able to be unifiable with the step below by throwing a `*` at the end of the `...wasm2obj}` for unix which should cover both Unix and Windows

Reply:
Unfortunately no: there are more files in there that start with these names, and that we don't want to copy, such as `wasmtime.d`, `wasmtime.dSYM`, and under some conditions `wasmtime-*`-like names.

I tinkered around with this a bit, and couldn't come up with a pattern that captured exactly the four names we want, without failing if not all are present. Suggestions very much welcome!